src/server/dfly_main.cc

```diff
@@ -456,103 +456,128 @@ bool ShouldUseEpollAPI(const base::sys::KernelVersion& kver) {
   return true;
 }
 
-io::Result<string> GetCGroupPath() {
+error_code GetCGroupPath(string* memory_path, string* cpu_path) {
+  CHECK(memory_path != nullptr);
+  CHECK(cpu_path != nullptr);
+
   // Begin by reading /proc/self/cgroup
 
   auto cg = io::ReadFileToString("/proc/self/cgroup");
   CHECK(cg.has_value());
```

Comment:
Since our process will die if it can't find / parse any of these files, I think that it can be useful for users to know *why* it died (and also for us supporting them :)
Could I ask that in all of these `CHECK()`s you'll add some info, for example here: `CHECK(cg.has_value()) << "Can't read /proc/self/cgroup";` ?
(I know you didn't add this specific piece of code, but since you're here anyways :)

Reply:
And to clarify, also in the `CHECK()`s below it would be useful

Reply:
Very cool, didn't know it is possible to pipe into `CHECK()`!